onnxruntime/core/providers/webnn/builders/impl/attention_op_builder.cc

```diff
@@ -0,0 +1,568 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
It'd be better to rename the file as `gqa_op_builder.cc` specific for `GQA` op, and extract common utils into an additional `attention_utils.cc` file for supporting other attention ops, e.g. `MHA`

Reply:
Please also add this op into `js/web/docs/webnn-operators.md` file.